src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,81 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static User validUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        validUser = new User("validLogin", "validPassword", 18);
+        registrationService.register(validUser);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidRegistrationDataException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        User userWithSameLogin = new User("validLogin", "anotherPassword", 19);
+        assertThrows(InvalidRegistrationDataException.class,
+                () -> registrationService.register(userWithSameLogin));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User userWithNullLogin = new User(null, "validPassword", 18);
+        assertThrows(InvalidRegistrationDataException.class,
+                () -> registrationService.register(userWithNullLogin));
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
```

Comment:
The test case should check for edge cases. If the minimum login length is 6, you should test with a login of exactly 6 characters to ensure it is accepted.